atomic_reactor/utils/operator.py

```diff
@@ -104,7 +236,31 @@ def description(self):
 
 
 class Annotation(NamedPullspec):
-    _image_key = NotImplemented
+    """
+    Annotation pullspecs are special, they may be found under any key
+    and there may be more than one pullspec under a single key
+    """
+
+    def __init__(self, data):
+        super(Annotation, self).__init__(data)
+        self._image_key = NotImplemented
+        self._start_i = NotImplemented
+        self._end_i = NotImplemented
+
+    @property
+    def image(self):
+        i, j = self._start_i, self._end_i
+        return self.data[self._image_key][i:j]
+
+    @image.setter
+    def image(self, value):
+        # If there are 2 or more pullspecs in the same annotation text,
+        # they *must* be replaced starting with the last one, otherwise
+        # replacing one pullspec would invalidate the (start, end) indices
+        # of the others. This is up to the code that uses this class.
```

Comment:
To clarify, the method `OperatorCSV.replace_pullspecs_everywhere` will use this class correctly?

Reply:
Any method that needs to replace annotations, so both `OperatorCSV.replace_pullspecs_everywhere` and the variant without `_everywhere` must use this correctly.

To further clarify, I'm *only* using the heuristic for replacing annotations, nothing else. So the `replace_pullspecs_everywhere` method now comes down to:
* Replace bare pullspecs (not embedded in text) everywhere except metadata.annotations
* Replace annotations using the heuristic

Reply:
Great!